src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -195,8 +196,12 @@ public override Server Server
         /// Add a backup file to restore plan media list
         /// </summary>
         /// <param name="filePaths"></param>
-        public void AddFiles(string filePaths)
+        public void AddDevices(string filePaths, DeviceType deviceType)
         {
+            if (isSupportedDeviceType(deviceType))
```

Comment:
Assuming that Url type is only getting supported for MI, where are we checking if the intended use is for MI or not?

Reply:
Good point. I'm not sure if we want to enable URL backup/restore only for MI.